custom_components/powercalc/strategy/linear.py

```diff
@@ -139,9 +145,5 @@ async def validate_config(self, source_entity: SourceEntity):
                 )
             )
 
-        if self._config.get(CONF_CALIBRATE) is None:
-            if self._config.get(CONF_MIN_POWER) is None:
-                raise StrategyConfigurationError("You must supply min power")
-
-            if self._config.get(CONF_MAX_POWER) is None:
-                raise StrategyConfigurationError("You must supply max power")
+        if not CONF_CALIBRATE in self._config and not CONF_MAX_POWER in self._config:
+            raise StrategyConfigurationError("You must supply max power")
```

Comment:
Well, linear mode documentation lists maximum power as optional:

https://github.com/bramstroker/homeassistant-powercalc#linear-mode

Reply:
One of `max_power` or `calibrate` is required, when you supply `calibrate` than `max_power` can be left empty, so that's why it is marked optional. Maybe this should be made clearer in the readme somehow. If you have any suggestions/improvements, PR's are welcome.